flipper/client.py

```diff
@@ -63,31 +88,45 @@ def list(
 
     @flag_must_exist
     def enable(self, feature_name: str):
+        self._event_emitter.emit(EventType.PRE_ENABLE, feature_name)
         self._store.set(feature_name, True)
+        self._event_emitter.emit(EventType.POST_ENABLE, feature_name)
 
     @flag_must_exist
     def disable(self, feature_name: str):
+        self._event_emitter.emit(EventType.PRE_DISABLE, feature_name)
         self._store.set(feature_name, False)
+        self._event_emitter.emit(EventType.POST_DISABLE, feature_name)
 
     @flag_must_exist
     def destroy(self, feature_name: str):
+        self._event_emitter.emit(EventType.PRE_DESTROY, feature_name)
         self._store.delete(feature_name)
+        self._event_emitter.emit(EventType.POST_DESTROY, feature_name)
 
     @flag_must_exist
     def add_condition(self, feature_name: str, condition: Condition):
         meta = FeatureFlagStoreMeta.from_dict(self.get_meta(feature_name))
 
         meta.conditions.append(condition)
 
+        self._event_emitter.emit(EventType.PRE_ADD_CONDITION, feature_name, condition)
```

Comment:
Why do some of these events use named arguments and others don't?

Reply:
It always matches the signature of the current client method